src/platform/plugins/shared/workflows_management/public/features/workflow_execution_detail/ui/execution_detail.tsx

```diff
@@ -0,0 +1,143 @@
+  workflowExecutionId,
+  setSelectedStepExecution,
+  selectedStepExecutionId,
+  setSelectedStep,
+  onClose,
+}) => {
+  const {
+    data: workflowExecution,
+    isLoading,
+    error,
+    refetch,
+  } = useWorkflowExecution(workflowExecutionId);
+
+  const closeFlyout = useCallback(() => {
+    setSelectedStepExecution(null);
+  }, [setSelectedStepExecution]);
+
+  useEffect(() => {
+    if (!workflowExecution) {
+      return;
+    }
+
+    const intervalId = setInterval(() => {
+      if (
+        ![
+          ExecutionStatus.COMPLETED,
+          ExecutionStatus.FAILED,
+          ExecutionStatus.CANCELLED,
+          ExecutionStatus.SKIPPED,
+        ].includes(workflowExecution.status)
+      ) {
+        refetch();
+        return;
+      }
+
+      clearInterval(intervalId);
+    }, 500); // Refresh every 500ms
+
+    return () => clearInterval(intervalId);
+  }, [workflowExecution, refetch]);
+
+  const renderSelectedStepExecutionFlyout = useCallback(() => {
+    if (!workflowExecution?.stepExecutions?.length) {
+      return null;
+    }
+
+    const selectedStepExecutionIndex = workflowExecution.stepExecutions.findIndex(
+      (step) => step.id === selectedStepExecutionId
+    );
+    const selectedStepExecution = workflowExecution.stepExecutions[selectedStepExecutionIndex];
+
+    if (!selectedStepExecution) {
+      return null;
+    }
+
+    const goNext =
+      selectedStepExecutionIndex < workflowExecution.stepExecutions.length - 1
+        ? () => {
+            const nextStepExecutionIndex =
+              (selectedStepExecutionIndex + 1) % workflowExecution.stepExecutions.length;
+            const nextStepExecution = workflowExecution.stepExecutions[nextStepExecutionIndex];
+            setSelectedStepExecution(nextStepExecution.id);
+          }
+        : undefined;
+
+    const goPrevious =
+      selectedStepExecutionIndex > 0
+        ? () => {
+            const previousStepExecutionIndex =
+              (selectedStepExecutionIndex - 1 + workflowExecution.stepExecutions.length) %
+              workflowExecution.stepExecutions.length;
+            const previousStepExecution =
+              workflowExecution.stepExecutions[previousStepExecutionIndex];
+            setSelectedStepExecution(previousStepExecution.id);
+          }
+        : undefined;
+
+    return (
+      <WorkflowStepExecutionFlyout
+        workflowExecutionId={workflowExecutionId}
+        stepExecutionId={selectedStepExecution.id}
+        stepExecution={selectedStepExecution}
+        closeFlyout={closeFlyout}
+        goNext={goNext}
+        goPrevious={goPrevious}
+        setSelectedStepId={setSelectedStep}
+        isLoading={isLoading}
+      />
+    );
+  }, [
+    workflowExecution?.stepExecutions,
+    workflowExecutionId,
+    closeFlyout,
+    setSelectedStep,
+    selectedStepExecutionId,
+    setSelectedStepExecution,
+    isLoading,
+  ]);
+
+  return (
+    <>
+      {renderSelectedStepExecutionFlyout()}
```

Comment:
Consider using `useMemo` instead of a callback, so it can be memoized properly:
```suggestion
      {selectedStepExecutionFlyout}
```